backend/tool/data.go

```diff
@@ -0,0 +1,58 @@
+package tool
+
+import (
+	"database/sql"
+
+	"github.com/short-d/app/fw/db"
+	"github.com/short-d/short/backend/app/usecase/keygen"
+)
+
+// Data transform existing data to the new format and move them to the correct
+// location.
+type Data struct {
+	keyGen keygen.KeyGenerator
+	db     *sql.DB
+}
+
+// EmailToID generates IDs for users and changes DB tables reference those IDs.
+func (d Data) EmailToID() {
+	rows, err := d.db.Query(`
```

Comment:
I suggest using batch reading

```
SELECT email FROM "user" where "id is empty" LIMIT 1000
```

Here "id is empty" means, that we fetch only records with unsettled `id`. We perform the above query again and again until after the select dataset size is less or equal to 1000.

It gives us a feeling to be sure that we can handle this migration even for a big number of records "just a good practice"

Reply:
@alldroll Should we automatically increase the number of counts using for loops or we give admin the control to decide the size?

Reply:
It depends but in general, we increase the number of counts. I think the simplest way is the best one:)

Reply:
@alldroll Done.